packages/web-forms/src/components/common/map/MapBlock.vue

```diff
@@ -28,41 +31,36 @@ const showErrorStyle = inject<ComputedRef<boolean>>(
 	computed(() => false)
 );
 
-const mapHandler = useMapBlock();
+const mapHandler = useMapBlock(props.mode, () => emitSavedFeature());
 
 onMounted(() => {
 	if (!mapElement.value || !mapHandler) {
 		return;
 	}
 
-	mapHandler.initializeMap(mapElement.value, props.featureCollection);
-	mapHandler.toggleClickBinding(!props.disabled);
-	mapHandler.setSavedByValueProp(props.savedFeatureValue);
+	mapHandler.initMap(mapElement.value, props.featureCollection, props.savedFeatureValue);
+	mapHandler.setupMapInteractions(props.disabled);
 	document.addEventListener('keydown', handleEscapeKey);
 });
 
 onUnmounted(() => {
 	document.removeEventListener('keydown', handleEscapeKey);
+	mapHandler.teardownMap();
 });
 
 watch(
 	() => props.featureCollection,
-	(newData) => {
-		mapHandler.loadGeometries(newData);
-		mapHandler.setSavedByValueProp(props.savedFeatureValue);
-	},
+	(newData) => mapHandler.updateFeatureCollection(newData, props.savedFeatureValue),
 	{ deep: true }
 );
 
-watch(
-	() => props.savedFeatureValue,
-	(newSaved) => mapHandler.setSavedByValueProp(newSaved)
-);
+watch(() => props.savedFeatureValue, mapHandler.setSavedByValueProp);
```

Comment:
I prefer the previous style because it's clearer that attributes are being passed through from the watch callback to setSavedByValueProp.

This pattern can also lead to really obscure bugs because it's implicitly also passing through other arguments.